src/Employer/Employer.Web/Controllers/Part1/TitleController.cs

```diff
@@ -53,10 +69,27 @@ public async Task<IActionResult> Title(TitleEditModel m, [FromQuery] bool wizard
                 vm.PageInfo.SetWizard(wizard);
                 return View(vm);
             }
-
+            var currentVacancy = await _orchestrator.GetCurrentVacancy(m,response.Data);
+            if (m.ReferredFromMAHome_FromSavedFavourites)
+            {
+                if (currentVacancy.TrainingProvider != null && !string.IsNullOrWhiteSpace(currentVacancy.ProgrammeId))
+                    return RedirectToRoute(RouteNames.NumberOfPositions_Get, new { vacancyId = response.Data });
+                if (currentVacancy.TrainingProvider == null)
+                    return RedirectToRoute(RouteNames.TrainingProvider_Select_Get, new { vacancyId = response.Data });
+            }
+                
             return wizard
                 ? RedirectToRoute(RouteNames.Training_Get, new {vacancyId = response.Data})
                 : RedirectToRoute(RouteNames.Vacancy_Preview_Get);
         }
+
+        private TitleEditModel GetReferredDataFromTempData(TitleEditModel m)
```

Comment:
I don't think the models need to know about this information can you remove these properties and just work this out in the controller.  The Models should only know about that they care about.  In this case its just the information required to show/hide and set properties of a back link.